### DISCUSSION TITLE
Help us improve WinUI - Quick survey

### DISCUSSION THREAD

**Discussion by Rinku-Stephen** [microsoft/microsoft-ui-xaml#9807] [General]
Your feedback matters! We have rolled out a brief survey with just 2 questions—it won't take much of your time.
Please share your thoughts on how we're doing and help us improve!

[Survey Link]No longer live
-- XAML/Winui maintainers

**Comment by Rinku-Stephen** [comment]
Hi everyone,

The survey will be live for another two weeks. Your feedback is crucial in helping us improve the WinUI experience. Please take a moment to share your thoughts with us.

Thank you!

**Comment by Rinku-Stephen** [comment]
Hi All - The survey will be live till end of the week. Please do fill the survey to let us know what's hurting your developer experience.

**Comment by Rinku-Stephen** [comment]
Thank you to the participants! We are processing your feedback and will get back soon with the findings and next steps. 

The survey is closed as of today. However, you can still share feedback as comments in this thread. We plan to run future surveys half yearly.

**Comment by Rinku-Stephen** [comment]
Hi everyone ! Thank you for all the feedback and responses! We, the WinUI team, have received ****58** responses** in all, and the survey is now closed.
The NSAT result, which calculates **net satisfaction, was 50 (out of a maximum 200)**.  This is our first survey for WinUI, and we see opportunities to improve and make the WinUI developer experience more delightful.

<html xmlns:o="urn:schemas-microsoft-com:office:office"
xmlns:w="urn:schemas-microsoft-com:office:word"
xmlns:m="http://schemas.microsoft.com/office/2004/12/omml"
xmlns="http://www.w3.org/TR/REC-html40">

<head>

<meta name=ProgId content=Word.Document>
<meta name=Generator content="Microsoft Word 15">
<meta name=Originator content="Microsoft Word 15">
<link rel=File-List
href="file:///C:/Users/RINKUS~1/AppData/Local/Temp/msohtmlclip1/01/clip_filelist.xml">

<link rel=themeData
href="file:///C:/Users/RINKUS~1/AppData/Local/Temp/msohtmlclip1/01/clip_themedata.thmx">
<link rel=colorSchemeMapping
href="file:///C:/Users/RINKUS~1/AppData/Local/Temp/msohtmlclip1/01/clip_colorschememapping.xml">




</head>

<body lang=EN-US style='tab-interval:36.0pt;word-wrap:break-word'>



Survey | NSAT
-- | --
NSAT (June ‘24   - this survey) | 50




</body>

</html>
38 responses included optional comments, which are summarized below:
<html xmlns:o="urn:schemas-microsoft-com:office:office"
xmlns:w="urn:schemas-microsoft-com:office:word"
xmlns:m="http://schemas.microsoft.com/office/2004/12/omml"
xmlns="http://www.w3.org/TR/REC-html40">

<head>

<meta name=ProgId content=Word.Document>
<meta name=Generator content="Microsoft Word 15">
<meta name=Originator content="Microsoft Word 15">
<link rel=File-List
href="file:///C:/Users/RINKUS~1/AppData/Local/Temp/msohtmlclip1/01/clip_filelist.xml">

<link rel=themeData
href="file:///C:/Users/RINKUS~1/AppData/Local/Temp/msohtmlclip1/01/clip_themedata.thmx">
<link rel=colorSchemeMapping
href="file:///C:/Users/RINKUS~1/AppData/Local/Temp/msohtmlclip1/01/clip_colorschememapping.xml">

<style>

</style>

</head>

<body lang=EN-US style='tab-interval:36.0pt;word-wrap:break-word'>



Top   Concerns | Total Samples
-- | --
Need for WinUI3 development, improved documentation, and   addressing tooling deficiencies. | 16
Calls for improved transparency, bug fixes, and more   responsive handling of bug reports. | 11
Desire for Open-Source Contributions and Community   Involvement | 5
Desire for WinUI3 to achieve feature parity with other   frameworks | 3
Need for Improved Management and Community Engagement | 3




</body>

</html>
<html xmlns:o="urn:schemas-microsoft-com:office:office"
xmlns:w="urn:schemas-microsoft-com:office:word"
xmlns:m="http://schemas.microsoft.com/office/2004/12/omml"
xmlns="http://www.w3.org/TR/REC-html40">

<head>

<meta name=ProgId content=Word.Document>
<meta name=Generator content="Microsoft Word 15">
<meta name=Originator content="Microsoft Word 15">
<link rel=File-List
href="file:///C:/Users/RINKUS~1/AppData/Local/Temp/msohtmlclip1/01/clip_filelist.xml">

<link rel=themeData
href="file:///C:/Users/RINKUS~1/AppData/Local/Temp/msohtmlclip1/01/clip_themedata.thmx">
<link rel=colorSchemeMapping
href="file:///C:/Users/RINKUS~1/AppData/Local/Temp/msohtmlclip1/01/clip_colorschememapping.xml">

<style>

</style>

</head>

<body lang=EN-US link="#467886" vlink="#96607D" style='tab-interval:36.0pt;
word-wrap:break-word'>


<p class=MsoNormal>As next steps, we plan to update the developer tooling
experience for WinUI with a designer and update the documentation. We will be
more active and transparent in communication with the community and resolve
bugs at the earliest. <o:p></o:p></p>

<p class=MsoNormal>As first steps, we are excited to share the bugs and
features we are working on and will bring to you at the earliest.<span
style='mso-tab-count:1'>             </span><o:p></o:p></p>


Feature | Issue Source
-- | --
Designer | ·         Discussion:   WinUI 3.0 XAML Designer · Issue #5917 · microsoft/microsoft-ui-xaml   (github.com)
WinUI3 Appx Bundles | ·         Warn   that Visual Studio does not produce appx bundles for a Winui3 Blank App   Packaged - Developer Community   ·         Visual   Studio misbehaving when trying to create App packages for distribution with   MS Store - Developer Community   ·         Bug   51291842: App launch fails when an App is deployed/packaged via MSIX - Boards   (visualstudio.com)
VS Installer search | ·         Introduce search for WinUI components in VS   installer components. Currently this is available in preview.
IntelliSense | ·       https://github.com/microsoft/microsoft-ui-xaml/issues/4199


<p class=MsoNormal>We will leave this post open for a while longer so that it
has more visibility, and you can continue to share your feedback as comments
here that we will action on. We plan to send the next survey Apr’25. <o:p></o:p></p>


</body>

</html>

**Reply by davepruitt** [reply]
I'm happy to see a designer is now on the list of priorities. Will this translate over to MAUI as well? Also: I would have loved to have taken the survey, but I didn't even know the survey existed.

**Reply by ozanyasindogan** [reply]
Wow that's great news, finally!

**Reply by MPITech** [reply]
Awesome news, sooo looking forward to the designer at last!

**Reply by luca-domenichini** [reply]
This is huge! Finally a designer is on roadmap!!!! 🚀

**Reply by JacobRusso** [reply]
Yes, it would be awesome if the XAML Designer also works for MAUI.

**Reply by Rinku-Stephen** [reply]
> I'm happy to see a designer is now on the list of priorities. Will this translate over to MAUI as well? Also: I would have loved to have taken the survey, but I didn't even know the survey existed.

Hi @davepruitt - Too bad that you missed the anonymous survey but feel free to share any feedback you are comfortable with being public as comments on this discussion and we will consider them for future prioritization...

**Reply by mdtauk** [reply]
Hopefully any new designer will work better than UWPs and Silverlight/WPFs - XCode's SwiftUI previews, and the best in class UI designers need to be a goal, even if it takes some iteration.

![image](https://github.com/user-attachments/assets/075f4ef6-6fd2-4d2d-ab50-b1221d4721be)

![image](https://blogger.googleusercontent.com/img/b/R29vZ2xl/AVvXsEjlABhyphenhyphen1mHYL0vNFyHJDZHx10aSEiNckq56Id8dsX7-XFk2Dk4O4xcp8xUtd-CmKKp55rM4CESCjOCpcnJgYcKfDqVNfXEV1BYcwksV4MTeou0Y3-KAk3MrtixvZrAQSE5Yp8caUD9sfrI/s1600/image7.gif)

**Reply by MCAleckson** [reply]
Hallelujah.  As a longtime contributor to most all the forum threads on the requirement for at least a visual previewer for the last 3 years, I'm so glad to see this.

Thank you Microsoft for doing the right thing here.

**Reply by MPITech** [reply]
Hopefully this causes companies like Syncfusion and DevExpress to finally kick WinUI control development to a new level when they see the commitment to WinUI that you guys have.

**Reply by Jimex** [reply]
This is truly an incredibly exciting piece of good news.

**Reply by trydalch** [reply]
Very happy about the news of a designer. 

We're currently maintaining two versions of our XAML (and many code behind files) so that we can use the WPF visual designer, hoping that WinUI gets a designer soon. 

What is the expected timeline?

**Comment by Rinku-Stephen** [comment]
There are few queries that we still have and the survey didn't answer. One such question is - How can templates to create WinUI projects be better? If you have any preferences, early thoughts or suggestions - Share here - https://github.com/microsoft/microsoft-ui-xaml/discussions/9921

**Comment by kmgallahan** [comment]
The way the survey was worded sounded like it was specifically asking how the GitHub repo (github.com/microsoft/microsoft-ui-xaml) could be improved, rather than WinUI itself.

I didn't have much to mention in that regard, so I skipped the survey. Wonder if that was the case for others...

**Reply by davepruitt** [reply]
What were the actual questions that were asked?

**Reply by kmgallahan** [reply]
From what I saw it was a single multiple choice question asking roughly: ~"How can we improve or how satisfied are you with 'microsoft/microsoft-ui-xaml'?", followed by a free-form text block.

**Reply by Rinku-Stephen** [reply]
Hi @kmgallahan - Thank you for the feedback. I understand the survey wording might have been confusing... We will update them in the next iteration to be clear :)

**Comment by gautambjain** [comment]
Many users here wish they had taken the survey. It would be great if you can reopen @Rinku-Stephen . You would have better results.

**Reply by Rinku-Stephen** [reply]
Hi @gautambjain 
Your feedback outside the survey too is important and I would request you to share them in the comments below. There will also be more opportunities for you to share your thoughts in future surveys or feedback sessions.

Unfortunately, the July'24 iteration of the DevSat survey is a timebound activity. We had to close the survey because we needed time to process the satisfaction score, qualitative feedback and make updates to our roadmap. This activity is now complete, and the team is working on the actionable. Reopening the survey would impact the deliverables as the goals we had aligned on internally from the feedback might shift with the new input.

**Reply by MPITech** [reply]
I would like to see an in-Visual Studio notification for future surveys shown to WinUI developers.  I, too, would have liked to participate.

**Reply by Rinku-Stephen** [reply]
@MPITech Feedback taken - we will try to send out in-VS notification in the next iteration of the survey.

**Comment by uixiu** [comment]
I'm facing significant challenges with previewing complex XAML changes in my WinUI project due to the unreliability of Hot Reload. Recompiling and restarting the application is a time-consuming process, especially considering WinUI's slower build times. Even minor modifications to XAML can take several minutes on my high-performance laptop, which is frustrating and interrupts my workflow. I've explored alternative solutions like Xaml Studio and one Visual Studio extension, but these approaches have limitations (1. need xaml change, 2. Freezing VS) and haven't provided a satisfactory solution. I'm seeking advice from the community on effective strategies for previewing XAML changes without relying on Hot Reload (which rarely works for me)

**Reply by Rinku-Stephen** [reply]
Hi @uixiu  We are working on helping with this issue bts. Unfortunately, It's turning out to be a challenging issue to solve so please hang on while we figure it out asap.

**Comment by myokeeh** [comment]
I would really like to be able to pick WinUI3 / WindowsAppSDK apps as Kiosk apps as easily as I can currently with UWP apps in Windows Settings: https://github.com/microsoft/WindowsAppSDK/issues/3642

**Reply by Rinku-Stephen** [reply]
Feedback taken and adding this to our consideration list for issues to fix..

**Comment by Rinku-Stephen** [comment]
Hi Everyone - As we take steps to improve your WinUi developer experience, I'm excited to announce that the enhanced search for WinUI components is available in preview now. Check out this post by @Snigdha011997  [here ](https://github.com/microsoft/microsoft-ui-xaml/discussions/9953)for more details.. Setting up the environment to create WinUI projects should be easier now.. Do try out the feature and share your feedback..

**Comment by NicholasChrzan** [comment]
If you could address memory issues like:
https://github.com/microsoft/microsoft-ui-xaml/issues/9960
https://github.com/microsoft/microsoft-ui-xaml/issues/7282

it would be a great help.

**Reply by Rinku-Stephen** [reply]
Sure @NicholasChrzan  - Adding this to our consideration set of issues to be fixed.

**Comment by user239393** [comment]
Does somebody know the timeline to develop the XAML desinger?
I will skip Maui and use WinUI3 instead.

Thanks

**Reply by luca-domenichini** [reply]
I hope to see something releaved at MS Ignite - nov 2024 with the release of .NET 9..

**Reply by shahryarse** [reply]
any news of xaml designer?

**Reply by qinzhaotong** [reply]
still nothing about the WinUI3 designer was found...

**Reply by JimmyRespawn** [reply]
No designer is a deal breaker.

**Reply by AlexanderBlackman** [reply]
The upcoming Uno designer might end up working.  It's basically WinUI 3 when it comes to Windows Desktop.

**Comment by trydalch** [comment]
How can we be notified of future surveys?
